library/lib_exec.ks

```diff
@@ -88,4 +96,59 @@ function evaluate_function
   set expression to function_name + "(" + expression + ")".
   unset _exec__param_list.
   return evaluate(expression).
+}
+
+function get_suffix {
+  parameter
+    structure, //the structure to get the suffix of
+    suffix,    //the suffix to get
+    parameter_list IS false. //if the suffix is a function call this is the list of parameters for the suffix
+  local filePath is "1:/_get_suffix" + suffix.
+  local logStr IS "global _evaluate_result is { parameter o. return o:" + suffix.
+  if parameter_list:istype("list") {
+    set filePath to filePath + parameter_list:length.
+    local separator is "(".
+    global _exec__param_list IS parameter_list.
+    local i IS 0.
+    until i >= parameter_list:length
+    {
+      set logStr to logStr + separator + "_exec__param_list[" + i + "]".
+      set separator to ", ".
+      set i to i + 1.
+    }
+    set logStr to logStr + ")".
+  }
+  set filePath to path(filePath + ".tmp").
+  log_run_del(logStr + ". }.",filePath).
+  local result is _evaluate_result:call(structure).
+  if defined unset _exec__param_list.
+  unset _evaluate_result.
+  return result.
+}
+
+function set_suffix {
+  parameter
+    structure, //the structure to set the suffix of
+    suffix,    //the suffix to set
+    val.       //the value to set the suffix to
+  local filePath is path("1:/_set_suffix" + suffix + ".tmp").
+  local logStr IS "global _evaluate_result is { parameter o,v. set o:" + suffix + " to v. }.".
+  log_run_del(logStr,filePath).
+  local result is _evaluate_result:call(structure,val).
+  if defined unset _evaluate_result.
```

Comment:
Should be
```suggestion
  if defined _evaluate_result unset _evaluate_result.
```

Reply:
I removed the check from this function as I could think of no way to make `get_suffix` recursive thus the global will always exist when this function is called.